### DISCUSSION TITLE
Alerts section

### DISCUSSION THREAD

**Discussion by testlorenzospa** [nextgenhealthcare/connect#5292] [Q&A] [answered]
Hello.

Is it possible to send an alert message email when a message HL7 is filtering and it returns a NACK message after a control on that message?
If yes, how?

Thanks.

**Comment by pacmano1** [comment, marked as answer]
Yes.

While the built in alert handler can alert on any error, what you probably want to do in the response transformer of the destination that is sending the message it process the HL7 response message manually, check for an error, and proceed accordingly.  This same pattern applies to destinations in general.    i.e. if you let mirth handle the error automatically, your response transformer does not fire so you cannot execute an action due to the error so you need to "manaully" process the response to a sent message.

So for your use case, that normally means setting your response transformer to raw, then serializing the raw message to HL7, then checking for the NACK and proceeding accordingly.

**Reply by pacmano1** [reply]
See https://github.com/nextgenhealthcare/connect/issues/2075, for a related discussion.
